Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -54,13 +54,14 @@ GEMGeometry* GEMGeometryBuilderFromDDD::build(const DDCompactView* cview, const
 
 GEMGeometry* GEMGeometryBuilderFromDDD::buildGeometry(DDFilteredView& fview, const MuonDDDConstants& muonConstants)
 {
-  std::cout << "Building the geometry service" << std::endl;
   LogDebug("GEMGeometryBuilderFromDDD") <<"Building the geometry service";
   GEMGeometry* geometry = new GEMGeometry();
 
   LogDebug("GEMGeometryBuilderFromDDD") << "About to run through the GEM structure\n" 
 					<<" First logical part "
 					<<fview.logicalPart().name().name();
+
+  std::map<std::string,std::tuple<int,float,float,float,float> > chamberPar;
```

Comment:
The requirement is to get dimensions from defined geometry of the chamber, the sensitive parts and eta sections. Some of these do not map to a part of DetId but the string specifies it uniquely. One thing could be done that to control these operations using a parameter. But the constructor does not have ParameterSet as argument.
________________________________
From: David Lange [notifications@github.com]
Sent: 28 March 2016 22:42
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cms-sw/cmssw] bsunanda:Phase2-gem12 Get geometry information of GEM detector (#13848)


In Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc<https://github.com/cms-sw/cmssw/pull/13848#discussion_r57630189>:

>    LogDebug("GEMGeometryBuilderFromDDD") <<"Building the geometry service";
>    GEMGeometry* geometry = new GEMGeometry();
>
>    LogDebug("GEMGeometryBuilderFromDDD") << "About to run through the GEM structure\n"
>                                       <<" First logical part "
>                                       <<fview.logicalPart().name().name();
> +
> +  std::map<std::string,std::tuple<int,float,float,float,float> > chamberPar;


@bsunanda<https://github.com/bsunanda> - why a string? can we use a DetId instead?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/13848/files/bed092daad21f9eb10ac05e20f1556f6d675de93#r57630189>